upstairs/src/lib.rs

```diff
@@ -364,6 +364,81 @@ pub fn deadline_secs(secs: f32) -> Instant {
         .unwrap()
 }
 
+/// Array of data associated with three clients, indexed by `ClientId`
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(transparent)]
+pub struct ClientData<T>([T; 3]);
+
+impl<T> std::ops::Index<ClientId> for ClientData<T> {
+    type Output = T;
+    fn index(&self, index: ClientId) -> &Self::Output {
+        &self.0[index.get() as usize]
+    }
+}
+
+impl<T> std::ops::IndexMut<ClientId> for ClientData<T> {
+    fn index_mut(&mut self, index: ClientId) -> &mut Self::Output {
+        &mut self.0[index.get() as usize]
```

Comment:
`std::ops::IndexMut` is the trait that allows for array-style assignment, e.g. doing
```rust
let mut data = ClientData::new(0);
data[ClientId(0)] = 10; // this calls index_mut under the hood!
```

It would be weird to call `index_mut` directly, but we have lots of code that uses array-style assignments (since many of these types used to be `Vec`).